### PR TITLE
Revert envoy v3 migration

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Reverts the PRs that constituted the initial move from envoy v2 to envoy v3 API.

# Context

We had initially approached the envoy API migration in a piecemeal fashion, as envoy [claims](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/versioning) to support both v2 and v3 resources over either v2 or v3 transport with ADS. This does not appear to be the case, and we found that envoy was serializing v3 resources as v2, which could (in theory) result in tricky bugs.

We will continue to v2 -> v3 migration, but on a long lived feature branch (on the order of weeks) and merge it into a 1.5.0-beta release when it is ready.

Our top priority is Gloo stability for the 1.4 release, and envoy still supports v2 until the end of the year (which is after we will release Gloo 1.5).

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make update-deps generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick multi input regression test flake, known https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=d0d35d58-5874-4cec-b3d1-785b8120b9f2

File: Makefile [hunk 1]
```diff
@@ -561,7 +561,6 @@ push-kind-images: docker
 	kind load docker-image $(IMAGE_REPO)/gloo-envoy-wasm-wrapper:$(VERSION) --name $(CLUSTER_NAME)
 	kind load docker-image $(IMAGE_REPO)/certgen:$(VERSION) --name $(CLUSTER_NAME)
 	kind load docker-image $(IMAGE_REPO)/access-logger:$(VERSION) --name $(CLUSTER_NAME)
-	kind load docker-image $(IMAGE_REPO)/sds:$(VERSION) --name $(CLUSTER_NAME)
```
Comment (EItanya) [inline Makefile:564]: This should stay

File: projects/gloo/api/v1/failover.proto [hunk 2]
```diff
@@ -26,12 +26,15 @@ import "gloo/projects/gloo/api/v1/ssl.proto";
...
+    Locality locality = 1;
+
     // PrioritizedLocality is an implicitly prioritized list of lists of `LocalityLbEndpoints`. The priority of each
     // list of `LocalityLbEndpoints` is determined by its index in the list.
-    repeated PrioritizedLocality prioritized_localities = 1;
+    repeated PrioritizedLocality prioritized_localities = 2;
 
     message PrioritizedLocality {
-        repeated LocalityLbEndpoints locality_endpoints = 2;
+        repeated LocalityLbEndpoints locality_endpoints = 1;
```
Comment (EItanya) [inline projects/gloo/api/v1/failover.proto:37]: We should not revert this

File: projects/sds/pkg/run/run.go [hunk 3]
```diff
@@ -2,56 +2,33 @@ package run
...
-
-func Run(
-	rootCtx context.Context,
-	sslKeyFile, sslCertFile,
-	sslCaFile,
-	sdsServerAddress string,
-	sdsServerFactories []sds_server.EnvoySdsServerFactory,
-) error {
-	ctx, cancel := context.WithCancel(rootCtx)
+func Run(ctx context.Context, sslKeyFile, sslCertFile, sslCaFile, sdsServerAddress string) error {
```
Comment (EItanya) [inline projects/sds/pkg/run/run.go:17]: All of the changes to the SDS server are still useful, and do not conflict with any of the other stuff if we want to keep it around

Comment (kdorosh) [reply projects/sds/pkg/run/run.go:17]: was planning on keeping these changes around on the feature branch (and thus bringing them back to master), but given that there was a PR before and after this that we are reverting, reverting here (and then bringing back) makes this whole PR a no merge conflict operation

**Pull request merged**